flytepropeller/pkg/controller/handler.go

```diff
@@ -230,7 +230,6 @@ func (p *Propeller) Handle(ctx context.Context, namespace, name string) error {
 	}
 
 	streak := 0
-	defer p.metrics.StreakLength.Add(ctx, float64(streak))
```

Comment:
Golang's defer semantics allows us to use this idiom with a slight change. From the [spec](https://go.dev/ref/spec#Defer_statements):
> Each time a "defer" statement executes, the function value and parameters to the call are [evaluated as usual](https://go.dev/ref/spec#Calls) and saved anew but the actual function is not invoked.

This means that we can define a function that closes over `streak` like so:
```
defer func() { p.metrics.StreakLength.Add(ctx, float64(streak)) }()
```

Reply:
Done. As you can probably tell I don't really know what I'm doing with golang :sweat_smile: